src/ocl/batchnormocl.cpp

```diff
@@ -949,8 +954,8 @@ void BatchNormBackward(Handle& handle,
             else
             {
                 variant    = 0;
-                xlocalsize = 1024;
-                xgridsize  = 1024 * c;
+                xlocalsize = 256;
```

Comment:
You have introduced a hazard here -- well expanded it. I guess it was always a hazard. 
xlocalsize = 256 which is `MIO_BN_GRP0` in the BWD pass.
The statement `if((n > 64) && (in_cstride > 160))` allows for in_cstride > 256 to pass through to variant 0 if n <=64.
where the preprocessor statement:
`#define MIO_BN_SEGTMP (MIO_BN_HW * (MIO_BN_GRP0 / MIO_BN_HW))`
produces a 0 for MIO_BN_SEGTMP
then, this statement passes the zero on:
`#define MIO_BN_SEGMENT ((MIO_BN_SEGTMP > MIO_BN_NHW) ? (MIO_BN_NHW) : (MIO_BN_SEGTMP))`
to this statement:
`#define MIO_BN_NLOOP ((MIO_BN_NHW + MIO_BN_SEGMENT - 1) / MIO_BN_SEGMENT)`
which is a divide by zero.
The goal in the original marco logic is produce a ceiling calculation to map the workitems across MIO_BN_NHW.

Reply:
may be we should limit (MIO_BN_GRP0 / MIO_BN_HW) to 1 if this division produces 0.  I will make this change.

Reply:
After this change, all these tests are passed for in_cstride > 256 to pass through to variant 0 if n <=64
root@miopen908-1:/testx/MIOpen/build# ./bin/MIOpenDriver bnormfp16 -m 1 -s 1 -n 64 -c 2048 -H 18 -W 18 -V 1 -F 0 -b 1 -i 1 -t 1 -w 0
MIOpenDriver bnormfp16 -m 1 -s 1 -n 64 -c 2048 -H 18 -W 18 -V 1 -F 0 -b 1 -i 1 -t 1 -w 0
stats: bnormb, 0, 84942848, 84942848, 0, 617.796096, 0.412480
GPU Kernel Min Time Backwards Batch Normalization Elapsed: 0.412480 ms
Backwards prop batch norm verification passed on dx.
Backwards prop batch norm verification passed on dscale.
Backwards prop batch norm verification passed on dbias.
Backwards Prop Batch Norm Verifies on CPU and GPU.
root@miopen908-1:/testx/MIOpen/build# ./bin/MIOpenDriver bnormfp16 -m 1 -s 1 -n 32 -c 2048 -H 18 -W 18 -V 1 -F 0 -b 1 -i 1 -t 1 -w 0
MIOpenDriver bnormfp16 -m 1 -s 1 -n 32 -c 2048 -H 18 -W 18 -V 1 -F 0 -b 1 -i 1 -t 1 -w 0
stats: bnormb, 0, 42475520, 42475520, 0, 727.316416, 0.175201
GPU Kernel Min Time Backwards Batch Normalization Elapsed: 0.175201 ms
Backwards prop batch norm verification passed on dx.
Backwards prop batch norm verification passed on dscale.
Backwards prop batch norm verification passed on dbias.
Backwards Prop Batch Norm Verifies on CPU and GPU.
root@miopen908-1:/testx/MIOpen/build# ./bin/MIOpenDriver bnormfp16 -m 1 -s 1 -n 32 -c 2048 -H 20 -W 20 -V 1 -F 0 -b 1 -i 1 -t 1 -w 0
MIOpenDriver bnormfp16 -m 1 -s 1 -n 32 -c 2048 -H 20 -W 20 -V 1 -F 0 -b 1 -i 1 -t 1 -w 0
stats: bnormb, 0, 52436992, 52436992, 0, 912.053440, 0.172480
GPU Kernel Min Time Backwards Batch Normalization Elapsed: 0.172480 ms
Backwards prop batch norm verification passed on dx.
Backwards prop batch norm verification passed on dscale.
Backwards prop batch norm verification passed on dbias.
Backwards Prop Batch Norm Verifies on CPU and GPU.